### ISSUE TITLE
Executing man in a code cell causes unexpected output

### ISSUE THREAD

**Issue by mojones** [takluyver/bash_kernel#50]
Running the `man` command in a code cell causes some, but not all, characters to be repeated in the output:

![selection_016](https://cloud.githubusercontent.com/assets/893512/16412496/6e2ddd9c-3d24-11e6-864a-f2adc33b0406.png)

Copying and pasting the output into a text editor reveals that it actually contains a bunch of backspace characters:

![selection_017](https://cloud.githubusercontent.com/assets/893512/16412518/9202b882-3d24-11e6-83ec-d9bcf98aefd5.png)

I have absolutely no idea what's going on here - is it maybe something to do with the output from `man` going through a pager before being displayed in jupyter? I'm using bash notebooks for teaching purposes, so it would be good to be able to illustrate the `man` command.

**Comment by takluyver** [comment]
I think it's something to do with the formatting of man pages - the doubled-up letters are bold when I view them in a terminal, and letters like `_O` are underlined. PAGER is set to `cat`, though, and I'd hope that would turn off such formatting.

**Comment by takluyver** [comment]
I think I understand what it's doing - it decides to do the formatting if stdout is a tty, regardless of what the pager is. We're using a tty so that programs give us e.g. coloured output, but we don't process the backspace characters. I'll open an issue on jupyter/notebook to fix this, because it also affects other kernels.

**Comment by mojones** [comment]
Ahh, good catch, I didn't spot the correlation with bold letters in the terminal. Would you mind dropping a link here to the jupyter issue so I can keep an eye on it?

**Comment by takluyver** [comment]
Here we go: jupyter/notebook#1572

**Comment by takluyver** [comment]
The fix for this got merged in the notebook.

**Comment by jseabold** [comment]
Seeing this with bash_kernel 0.6 from pypi and jupyter 4.3.0 and jupyter notebook 5.0.0 on OS X 10.11.6.

Output of `man ls`.

```
LS(1)                     BSD General Commands Manual                    LS(1)

NNAAMMEE
     llss -- list directory contents

SSYYNNOOPPSSIISS
     llss [--AABBCCFFGGHHLLOOPPRRSSTTUUWW@@aabbccddeeffgghhiikkllmmnnooppqqrrssttuuwwxx11] [_f_i_l_e _._._.]

DDEESSCCRRIIPPTTIIOONN
     For each operand that names a _f_i_l_e of a type other than directory, llss
```

**Comment by jseabold** [comment]
It renders as `NNAAMMEE`, etc. in the notebook.

**Comment by takluyver** [comment]
It was meant to be fixed in notebook 5.0 - if you can definitely reproduce it there, please file a notebook issue referencing the one linked above.
